src/spineml/simulator/logOutput.cc

```diff
@@ -274,16 +233,73 @@ void SpineMLSimulator::LogOutput::AnalogueNetwork::record(double, unsigned int t
     }
 }
 
+//----------------------------------------------------------------------------
+// SpineMLSimulator::LogOutput::AnalogueNetwork
+//----------------------------------------------------------------------------
+SpineMLSimulator::LogOutput::AnalogueNetwork::AnalogueNetwork(const pugi::xml_node &node, double dt,
+                                                              const std::string &port, unsigned int popSize,
+                                                              const filesystem::path &logPath,
+                                                              const ModelProperty::Base *modelProperty)
+    : AnalogueExternal(node, dt, port, popSize, logPath, modelProperty)
+{
+    // Check size determined by indices/population size matches attribute
+    const unsigned int size = getIndices().empty() ? popSize : (unsigned int)getIndices().size();
+    //assert(size == node.attribute("size").as_uint());
+
+    // Allocate output buffer
+    m_OutputBuffer.resize(size);
+
+    // Read connection stats
+    const std::string connectionName = node.attribute("name").value();
+    const std::string hostname = node.attribute("host").value();
+    const int tcpPort = node.attribute("tcp_port").as_int();
+
+    LOGD << "\tAnalogue network log '" << connectionName << "' (" << hostname << ":" << tcpPort << ")";
+
+    // Attempt to connect network client
+    if(!m_Client.connect(hostname, tcpPort, size, NetworkClient::DataType::Analogue,
+        NetworkClient::Mode::Source, connectionName))
+    {
+        throw std::runtime_error("Cannot connect network client");
+    }
+}
+//----------------------------------------------------------------------------
+void SpineMLSimulator::LogOutput::AnalogueNetwork::recordInternal()
+{
+    // If no indices are specified, transform all values in model property into double precision
+    // **TODO** once precision is switchable this could be optimised out
+    if(getIndices().empty()) {
+        std::transform(getModelPropertyHostStateVarBegin(), getModelPropertyHostStateVarEnd(), m_OutputBuffer.begin(),
+                    [](scalar x)
+                    {
+                        return static_cast<double>(x);
+                    });
```

Comment:
Wouldn't ``std::copy`` do here? I *think* it casts values to the output type

Reply:
hmmm that would be cleaner-looking but it's very unclear to me from the documentation whether it's the case. It suggests that for trivial types, it'll do block copies using memmove but nothing about different types...

Reply:
I'll do a ``memmove`` if e.g. you're copying ``int``s from one array to another, but I think otherwise it'll just use the assignment operator, so you'll get an implicit cast. It's true that the documentation doesn't seem to mention this explicitly though.